binderhub/tests/test_repoproviders.py

```diff
@@ -171,7 +171,7 @@ async def test_hydroshare_doi():
         [
             "10.7910/DVN/TJCLKP",
             "10.7910/DVN/TJCLKP",
-            "3035124.v3.0",
+            "3035124.v4.0",
```

Comment:
@Xarthisius @pdurbin You were involved in the PR that added support for Dataverse and this test in
https://github.com/jupyterhub/binderhub/pull/969

Is this change in the resolved reference expected? Is there a way to make this constant so it doesn't change again in future?

Reply:
Is "10.7910/DVN/TJCLKP" associated with v3 or the latest version? Now we get the latest version from that as a reference given to binderhub, when we may or may not have expected we would get v3 even if there was a newer version.

Reply:
@consideRatio I'm not sure I'd put it quite that way.

"10.7910/DVN/TJCLKP" is associated with all versions of the dataset, including the latest version and previous versions (v3, v2, etc). Dataverse doesn't mint DOIs for each version of a dataset (though this functionality has certainly been [requested](https://github.com/IQSS/dataverse/issues/4499) for quite a while).

I still think `:latest-published` (above) might help but there's also a Dataverse API to list each version of a dataset, if that's of interest: https://guides.dataverse.org/en/5.12.1/api/native-api.html#list-versions-of-a-dataset

Sorry if I'm not quite following. If you're getting the latest version from "10.7910/DVN/TJCLKP" I'm not sure why v3 is hard coded (and updated to v4 in this PR).

Reply:
https://github.com/IQSS/dataverse/issues/4499 explains it, I assumed Dataverse supported exact DOIs resolving to a particular version for reproducibility 😞 .

I can think of two options here (I'm happy with whatever you prefer):
- Change this to `v4` as in the PR
- Change this (and the test case below) to a regex match for `v\d+`

Either way, adding a comment to this test case explaining all this would be helpful for future readers!

Reply:
Regex added and I've included a comment based on your comment here @pdurbin - thank you!!